Madness/String.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  String.swift
+//  Madness
+//
+//  Created by Josh Vera on 10/19/15.
+//  Copyright © 2015 Rob Rix. All rights reserved.
+//
+
+import Foundation
+
+public typealias CharacterParser = Parser<String.CharacterView, Character>.Function
+public typealias StringParser = Parser<String.CharacterView, String>.Function
+public typealias DoubleParser = Parser<String.CharacterView, Double>.Function
+
+public let double: DoubleParser = {
+	let minus = (%"-")|?
+	let digits = digit+
+	
+	return minus >>- { sign in
+		digits >>- { digits in
+			let double = digits.reduce(0.0, combine: { num, char in
+				return 10 * num + Double(String(char))!
```

Comment:
Addressed in https://github.com/robrix/Madness/commit/122583e5735ac0b59d307f8083bc06d60ad206ff